### ISSUE TITLE
Search functionality in stock.picking

### ISSUE THREAD

**Issue by echeverrifm** [odoo/odoo#5366]
** Search functionality in stock.packing **

Impacted versions:

 - 8.0

Steps to reproduce:

 1. We go to Warehouse/All Operations ( PACK Operations )
 2. We find some picking like ( WH\PACK\00003 ) 

Video of issue:

http://youtu.be/WLfSEVUGo5E

Current behavior:

 - Don't show and find any picking with this reference ( WH\PACK\00003 )

Expected behavior:

 - We expected a picking with this reference ( WH\PACK\00003 )

**Comment by nhomar** [comment]
Hypothesis comprobed.

If we seach 
    ( WH\\PACK\\00003 ) << see escaped chars it works.

What I think is that the char '\' should not be escaped.

**Comment by nim-odoo** [comment]
Hello,

Here is a pull request which should solve the issue: https://github.com/odoo/odoo/pull/5635

Regards

**Comment by odony** [comment]
Hi, I'm sorry we've apparently spent a lot of time on this issue already, but this is actually a feature, not a bug, and you should have been informed about it from the start.

tl;dr:
> We cannot at the same time allow SQL meta characters *and* allow searching on literal
> backslash characters. Like all search interfaces (like google, etc.) the users need to learn
> how to use it to achieve what they want.

Long answer:
By default Odoo supports SQL meta-characters in any place where a "contains" (`ilike`) search is performed. This is the case in the search view and in many other places.

This means you can use an underscore (`_`) character to search for any character, or a percent character (`%`) to search for any series of character. This can be very useful in some circumstances.
Because of this feature, you may also need to use a backslash character (`\`) to escape a real underscore or percent character if you want it to match literally. For example if you want to search for a record with "`50%`" in the name, you will need to search for "`50\%`" otherwise you will get all the records that contain "50", which is not what you want.

This is why, as @nhomar noted, you have to double the backslash characters when searching for "`WH\\PACK\\00003`" by default.
However you **can perfectly** search for "`WH\PACK\00003`" without doubling the backslashes if you use the advanced search with the "is equal to" operator instead of the default "contains".

So yes, like all search interfaces (google, etc.) the users need to learn how to use it properly to get the results they want.

However, some workarounds exist if you don't think the users will understand the principle of using a double backslash when their search contain a backslash, for example:
 - Don't use backslashes in document names and references, use forward slashes or other similar characters. Backslashes can cause many other problems including platform-specific issues
 - Modify the default search view for your users, and replace the default (first) field/filter to explicitly use a `filter_domain` with the "equal to" operator for the field in which you are using backslashes

Thanks for your understanding.
